autonomi/src/wallet.rs

```diff
@@ -0,0 +1,139 @@
+    let file_path = create_file_with_keys(wallet_private_key, wallet_public_key)?;
+    // println!("A file is created with the path: {}", file_path);
+    Ok(file_path)
+}
+
+pub fn encrypt_evm_wallet(file_path: String, password: String) -> Result<String,EvmError> {
+    if wallet_encryption_status(Path::new(&file_path)) {
+        return Ok(String::from("Provided File is already encrypted")); //replace with error condition later. 
+    }
+
+    let private_key = std::fs::read_to_string(&file_path)
+                                .map_err(|_| EvmError::OperationError("Not able to get the file dir path".to_string()))?;
+
+    let encrypted_private_key = encrypt_secret_key(&private_key, &password)?;
+       //make the wallet a directory.
+
+    if Path::new(&file_path).is_file() {
+        std::fs::remove_file(&file_path).expect("not able to remove the file");
+        std::fs::create_dir(&file_path).expect("not able to create the directory"); 
+    }
+
+    wallet_encryption_storage(&file_path, &encrypted_private_key)
+}
+
+
+
+pub fn get_private_key_from_wallet(key: u32, files: HashMap<u32, (String,String)>) -> Result<String, EvmError> {
+
+    match files.get(&key) {
+        Some(value) => {
+            let mut wallet_directory = get_wallet_directory();
+            wallet_directory.push(value.1.clone());
+            if value.0 == "unprotected" {
+                let file_contents= std::fs::read(&wallet_directory);
+                if let Ok(file_data) = file_contents {
+                    let private_key = String::from_utf8(file_data).map_err(|_| EvmError::OperationError("Not able to convert file contents".to_string()))?;
+                    
+                    return Ok(private_key);
+                }
+            }
+
+            if value.0 =="passw-protected" {
+                let _ = wallet_directory.push(ENCRYPTED_MAIN_SECRET_KEY_FILENAME);
+                println!("encrypted wallet path: {:?}", wallet_directory);
+                let encrypted_bytes = std::fs::read(wallet_directory);
+                if let Ok(file_data) = encrypted_bytes {
+                    let encrypted_private_key = String::from_utf8(file_data).map_err(|_| EvmError::OperationError("Not able to convert file contents".to_string()))?;
+                    let password = prompt_the_user_for_password();
+                    match password {
+                        Some(value) => {
+                            println!("Recieved the password, initiating decryption!");
+                            let private_key = decrypt_secret_key(&encrypted_private_key, &value)?;
+                            return Ok(private_key);
+                        },
+                        None => {
+                            println!("The provided Password is not proper, can not proceed further");
+                        }
+                    }
+                    
+                }
+            }
+        },
+        None => {
```

Comment:
Let's try to keep reasonable indentation level. I try to keep it under 3 levels.